api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -106,8 +108,18 @@
 
   @Override
   public List<PublicInstitutionDetails> getPublicInstitutionDetails() {
+    Map<Long, MembershipRequirement> registeredTierRequirementMap =
+        StreamSupport.stream(institutionTierRequirementDao.findAll().spliterator(), false)
+            .filter(t -> t.getAccessTier().getShortName().equals(REGISTERED_TIER_SHORT_NAME))
+            .collect(
+                Collectors.toMap(
+                    t -> t.getInstitution().getInstitutionId(),
+                    DbInstitutionTierRequirement::getMembershipRequirement));
     return StreamSupport.stream(institutionDao.findAll().spliterator(), false)
-        .map(publicInstitutionDetailsMapper::dbToModel)
+        .map(
+            i ->
+                publicInstitutionDetailsMapper.dbToModel(
+                    i, registeredTierRequirementMap.get(i.getInstitutionId())))
```

Comment:
This is fine, but are these not directly joinable in the hibernate model? were there issues with that?

Reply:
I followed EmailDomains and EmailsAddresses. And saw Joel had a PR switching from join in hibernate to this pattern:
https://github.com/all-of-us/workbench/pull/3118 (saying this is a better model)
I checked with him, he forgot the reason, and suggest me to ask Brain. So I just followed that, and forgot to check with Brain...
I can make it this way, and refactor it when I got the answer from him